pyhms/demes/initialize.py

```diff
@@ -3,61 +3,100 @@
     CMALevelConfig,
     DELevelConfig,
     EALevelConfig,
-    LHSLevelConfig,
     LocalOptimizationConfig,
-    SobolLevelConfig,
+    RandomLevelConfig,
 )
+from pyhms.utils.parameter_calculation import get_default_mutation_std
 from structlog.typing import FilteringBoundLogger
 
 from ..core.individual import Individual
+from ..core.initializers import (
+    GaussianInitializer,
+    GaussianInitializerWithSeedInject,
+    InjectionInitializer,
+    LHSGlobalInitializer,
+    PopInitializer,
+    SobolGlobalInitializer,
+    UniformGlobalInitializer,
+)
 from .abstract_deme import AbstractDeme
 from .cma_deme import CMADeme
 from .de_deme import DEDeme
 from .ea_deme import EADeme
-from .lhs_deme import LHSDeme
 from .local_deme import LocalDeme
-from .sobol_deme import SobolDeme
+from .random_deme import RandomDeme
 
 
 def init_root(config: BaseLevelConfig, logger: FilteringBoundLogger) -> AbstractDeme:
-    return init_from_config(config, "root", 0, 0, None, logger)
+    return init_from_config(config, "root", 0, 0, None, None, logger)
+
+
+def create_initializer(
+    config: BaseLevelConfig, sprout_seed: Individual | None, injected_population: list[Individual] | None
+) -> PopInitializer:
+    match config.pop_initializer_class.__qualname__:
```

Comment:
Let's assume that I have a very custom initialization scheme that is problem specific. How can I use it without editing this function?

Reply:
I implemented custom initialization for DistributionOptimization and created a draft PR: https://github.com/agh-a2s/pyhms/pull/79/files. The concept is straightforward: a new parameter in the configuration that specifies a function to create a new individual.

Reply:
Yes, this would require editing a function to add a custom initialization scheme to the match-case. I think, that your new solution is better.

Reply:
I will merge simple-initialization to this branch and then resolve conflicts with main

Reply:
I went through it again and used the concept of including the initializer type in config. In my opinion it still had some issues: you still had to deduce some initialization properties during deme initialization based on some run context, which resulted in decision trees in SHADEDeme init.